components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -129,6 +148,24 @@ public boolean canHandle(HttpServletRequest request) {
         return false;
     }
 
+
+    private boolean isInitialRequest(AuthenticationContext context, HttpServletRequest request) {
+
+        return !context.isLogoutRequest() && !hasCodeParamInRequest(request) && !hasErrorParamInRequest(request);
+    }
+
+    private boolean hasErrorParamInRequest(HttpServletRequest request) {
+
+        String error = request.getParameter(OIDCAuthenticatorConstants.OAUTH2_ERROR);
```

Comment:
We do not need this variable right?

Reply:
Fixed with 942f1989b7db43df06a41c8904171bef68e692b6